framework/auth/views.py

```diff
@@ -941,7 +943,7 @@ def external_login_email_post():
 
     external_id_provider = session.data['auth_user_external_id_provider']
     external_id = session.data['auth_user_external_id']
-    fullname = session.data['auth_user_fullname']
+    fullname = session.data.get('auth_user_fullname', form.name.data)
```

Comment:
Session data always has the key `auth_user_fullname` even if the value is empty. So the `.get()` fails to take `form.name.data` as default.

![screen shot 2018-06-21 at 10 11 17 am](https://user-images.githubusercontent.com/3750414/41725260-5e4afa88-753d-11e8-875c-65446835cc98.png)

Here is my update:

```python
fullname = session.data.get('auth_user_fullname', None)
if not fullname:
    fullname = form.name.data
```

Or maybe this looks better but not sure if `or` works for empty string:

```python
fullname = session.data.get('auth_user_fullname', None) or form.name.data
```

Reply:
More update:

```python
fullname = u'{} {}'.format(cas_resp.attributes.get('given-names', ''), cas_resp.attributes.get('family-name', '')).strip()
```
The `.strip()` guarantees that `fullname` is and empty string if no names exist. So the `or` should work.

```bash
Python 2.7.12 (default, Jul 19 2016, 01:27:03)
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> x = '' or 'test'
>>> x
'test'
```